### PR TITLE
Feat: alibaba provider addon

### PR DESCRIPTION
**Pull request by chivalryq**
**What this PR does / why we need it**:

Add first provider for terraform controller addon

usage:

```shell
vela addon enable terraform/provider-alibaba ALICLOUD_ACCESS_KEY=xxx ALICLOUD_SECRET_KEY=yyy ALICLOUD_SECURITY_TOKEN=zzz
```

Ref: https://github.com/oam-dev/kubevela/discussions/2173

### CODE REVIEW

File: charts/vela-core/templates/addons/terraform-provider-alibaba.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        status:
+          rollout:
+            batchRollingState: ""
+            currentBatch: 0
+            lastTargetAppRevision: ""
+            rollingState: ""
+            upgradedReadyReplicas: 0
+            upgradedReplicas: 0
+    status:
+      observedGeneration: 0
```
Comment (zzxwill) [inline charts/vela-core/templates/addons/terraform-provider-alibaba.yaml:39]: Are there fields necessary?

File: charts/vela-core/templates/addons/terraform-provider-alibaba.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+              apiVersion: v1
+              kind: Secret
+              metadata:
+                name: alibaba-provider-secret
+                namespace: vela-system
+              stringData:
+                credentials: |
+                  accessKeyID: [[ index .Args "ak" ]]
+                  accessKeySecret: [[ index .Args "sk" ]]
+                  securityToken: [[ index .Args "sts" ]]
```
Comment (zzxwill) [inline charts/vela-core/templates/addons/terraform-provider-alibaba.yaml:27]:
```suggestion
                  accessKeyID: [[ index .Args "ALICLOUD_ACCESS_KEY" ]]
                  accessKeySecret: [[ index .Args "ALICLOUD_SECRET_KEY" ]]
                  securityToken: [[ index .Args "ALICLOUD_SECURITY_TOKEN" ]]
```

Terraform users is family with those changed variables.

File: charts/vela-core/templates/addons/terraform-provider-alibaba.yaml (hunk 1 above)
Comment (chivalryq) [reply charts/vela-core/templates/addons/terraform-provider-alibaba.yaml:39]: generated when Marshual application to yaml, need deleted?

**Comment by zzxwill** [comment]
Where will a Provider object will be applied?

File: charts/vela-core/templates/addons/terraform-provider-alibaba.yaml (hunk 1 above)
Comment (zzxwill) [reply charts/vela-core/templates/addons/terraform-provider-alibaba.yaml:55]: Then feel free to delete it.

**Comment by zzxwill** [comment]
Can you run an application to verify this feature?

https://github.com/oam-dev/kubevela/blob/master/docs/examples/terraform/cloud-resource-provision-and-consume/application.yaml

**Comment by zzxwill** [comment]
@chivalryq Did the test passed?

**Comment by chivalryq** [comment]
> @chivalryq Did the test passed?

Yes!

Step:

1. vela addon enable terraform
2. vela addon enable terraform/provider-alibaba ALICLOUD_ACCESS_KEY=xxx ALICLOUD_SECRET_KEY=yyy ALICLOUD_SECURITY_TOKEN=zzz
3. Bang! Just use component related with aliyun in application

**Review by wonderflow — APPROVED** [review]

**Pull request merged**